src/guide/receive_payment.md

```diff
@@ -1,8 +1,177 @@
-# Receiving Lightning payments
+# Receiving Payments
 
 With the Breez Liquid SDK you aren't required to open a channel and set up your inbound liquidity.
 
-Once the SDK is initialized, you can directly begin receiving payments.
+Once the SDK is initialized, you can directly begin receiving payments. The receive process takes two steps:
+1. [Preparing the Payment](receive_payment.md#preparing-payments)
+1. [Receiving the Payment](receive_payment.md#receiving-payments-1)
+
+## Preparing Payments
+During the prepare step, the SDK ensures that the inputs are valid with respect to the specified payment method,
+and also returns the relative fees related to the payment so they can be confirmed.
+
+
+The SDK currently supports three methods of receiving: Lightning, Bitcoin and Liquid:
+
+### Lightning
+When receiving via Lightning, we generate an invoice to be paid.  Note that the payment may fallback to a direct Liquid payment (if the payer's client supports this).
+
+
+**Note:** The amount field is currently mandatory when paying via Lightning.
+
+<custom-tabs category="lang">
+<div slot="title">Rust</div>
+<section>
+
+```rust,ignore
+{{#include ../../snippets/rust/src/receive_payment.rs:prepare-receive-payment-lightning}}
+```
+</section>
+
+<div slot="title">Swift</div>
+<section>
+
+```swift,ignore
+{{#include ../../snippets/swift/BreezSDKExamples/Sources/ReceivePayment.swift:prepare-receive-payment-lightning}}
+```
+</section>
+
+<div slot="title">Kotlin</div>
+<section>
+
+```kotlin,ignore
+{{#include ../../snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ReceivePayment.kt:prepare-receive-payment-lightning}}
+```
+</section>
+
+<div slot="title">React Native</div>
+<section>
+
+```typescript
+{{#include ../../snippets/react-native/receive_payment.ts:prepare-receive-payment-lightning}}
+```
+</section>
+
+<div slot="title">Dart</div>
+<section>
+
+```dart,ignore
+{{#include ../../snippets/dart_snippets/lib/receive_payment.dart:prepare-receive-payment-lightning}}
+```
+</section>
+</custom-tabs>
+
+### Bitcoin
+When receiving via Bitcoin, we generate a Bitcoin BIP21 URI to be paid.
```

Comment:
Isn't this section basically identical to `receive_onchain.md`? Here we split `prepare` and `receive_payment` in two snippets, one here and one at the bottom of the page. In `receive_onchain.md` we have both essentially in one section of the page.

I wonder if it's not better to simply link to `receive_onchain.md` under this Bitcoin section, as that page has more details (refund, rescan).

What do you think?

Reply:
Was discussing this with Erdem. For the sake of semplicity, I duplicated, but it's also possible to link to the other section. What do you think @dangeross?

Reply:
I'm struggling with it because the title is `Receiving an on-chain transaction` but is missing Liquid onchain.
I think I would prefer to:
- have only the receive_payments page / remove receive_onchain page
- add a refund_swap sub page (e.g. 3.1.1 Refund a swap)
- add a rescanning page (e.g. 3.4 Rescanning swaps)
- add a "Developer note" under the Bitcoin section to a refund_swap sub page